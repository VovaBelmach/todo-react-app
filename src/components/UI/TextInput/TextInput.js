import React from "react";
import PropTypes from "prop-types";
import styles from "./TextInput.module.css";

const TextInput = (props) => { 
  return (
    <>
      <input
        type="text"
        className={styles.textInput}
        name={props.name}
        value={props.value}
        placeholder={props.placeholder}
        onChange={props.onChangeHandler}
        onBlur={props.onBlurHandler}
      />
    </>
  );
};

TextInput.propTypes = {
  name: PropTypes.string,
  value: PropTypes.any,
  placeholder: PropTypes.string,
  onChangeHandler: PropTypes.func,
  onBlurHandler: PropTypes.func,
};

export default TextInput;
