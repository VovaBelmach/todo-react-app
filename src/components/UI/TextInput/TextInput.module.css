.textInput {
    font: inherit;
    display: block;
    width: 90%;
    padding: 0.15rem;
    border: none;
    outline: none;
}

.textInput:focus {
    outline: none;
    border-color: #4f005f;
}