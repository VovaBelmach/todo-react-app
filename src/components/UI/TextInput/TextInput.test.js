import React from "react";
import renderer from "react-test-renderer";
import TextInput from "./TextInput";

describe("TextInput", () => {
  it("should renders correctly", () => {
    const tree = renderer.create(<TextInput name="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
