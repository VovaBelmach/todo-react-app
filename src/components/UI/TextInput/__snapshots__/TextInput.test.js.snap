// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should renders correctly 1`] = `
<input
  className="textInput"
  name="test"
  type="text"
/>
`;
